src/main/create-operator-json/core/operatorFactory.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+/** 
+ * Base class,
+ * use this to create new Operator class objects
+*/
+export default class Operator{
+
+    #id; // unique operator integer, 
+    #name; // operators name, 
+    #side; // defending or attacking, 
+    #organization; // SEALs, GIGN, etc, 
+    #operation; // operation release title, N/A for core operators
+    #counters; // an array of objects, used by D3.js for connecting operator nodes
+
+    /** 
+     * 
+     * @param {object} person object with operator info, Use an object import of r6operatos node_module
+     * @param {number} id represents unique operator id, Use operatorIdEnum.js to fill this out
+     * @param {string} operation name of the operation operator was released on, N/A if one of the core 17 operators
+     */
+    constructor(person, id, operation) {
+        if(!person || !id || !operation) {
+            throw 'Constructor is missing a field'
+        }
+        if(typeof person !== 'object' || typeof id !== 'number' || typeof operation != 'string') {
```

Comment:
Your docs and error handling here is just fantastic. Superb work :)

Reply:
Thank you :)